From fad2e50ca6a95c572b4d1f57aa44f20692681029 Mon Sep 17 00:00:00 2001
From: Shawn Thiah <shawn.thiah@kape.com>
Date: Thu, 26 Mar 2026 11:42:43 +0800
Subject: [PATCH] CVPN-1945 Lower max mtu for DTLS 1.3 handshake messages to
 prevent fragmentation

---
 src/dtls13.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/dtls13.c b/src/dtls13.c
index 5af638645..90662d24c 100644
--- a/src/dtls13.c
+++ b/src/dtls13.c
@@ -990,7 +990,9 @@ static int Dtls13SendFragmentedInternal(WOLFSSL* ssl)
     isEncrypted = Dtls13TypeIsEncrypted(
         (enum HandShakeType)ssl->dtls13FragHandshakeType);
     rlHeaderLength = Dtls13GetRlHeaderLength(ssl, isEncrypted);
-    maxFragment = wolfssl_local_GetMaxPlaintextSize(ssl);
+    // Pete M: Removing 500 bytes to ensure the packet fragments at a ratio
+    //         of 70/30 rather than 99/1 
+    maxFragment = wolfssl_local_GetMaxPlaintextSize(ssl) - 500;
     if (maxFragment <= DTLS_HANDSHAKE_HEADER_SZ ||
             maxFragment > MAX_RECORD_SIZE ||
             ssl->dtls13FragOffset > ssl->dtls13MessageLength) {
-- 
2.53.0

